xc/domain.ml

```diff
@@ -1791,6 +1791,11 @@ let soft_reset ~xc ~xs domid =
   xs.Xs.write (dom_path ^ "/store/port") (string_of_int store_port) ;
   xs.Xs.write (dom_path ^ "/console/port") (string_of_int console_port) ;
   Xenctrlext.domain_update_channels xc domid store_port console_port ;
+  (* reset PV features and disengage balloon driver *)
+  List.iter (fun p -> xs.Xs.rm (dom_path ^ "/control/feature-" ^ p))
```

Comment:
The code looks good. If we want to keep the string construction in one place, this would also work:

```
List.iter (fun p -> Printf.sprintf "%s/control/feature-%s" dom_path p |> xs.Xs.rm)
   ["suspend"; "poweroff"; "reboot"; "vcpu-hotplug"; "balloon"]
```

Reply:
What if a path doesn't exist? This may be the case for feature paths. Perhaps wrap the call with `log_exn_continue`.